modules/percolator/src/main/java/org/elasticsearch/percolator/ExtractQueryTermsService.java

```diff
@@ -243,48 +419,38 @@ public static void extractQueryTerms(Query query, ParseContext.Document document
         }
     }
 
-    private static int minTermLength(Set<Term> terms) {
+    static int minTermLength(Set<Term> terms) {
         int min = Integer.MAX_VALUE;
         for (Term term : terms) {
             min = Math.min(min, term.bytes().length);
         }
         return min;
     }
 
-    /**
-     * Creates a boolean query with a should clause for each term on all fields of the specified index reader.
-     */
-    public static Query createQueryTermsQuery(IndexReader indexReader, String queryMetadataField,
-                                              String unknownQueryField) throws IOException {
-        Objects.requireNonNull(queryMetadataField);
-        Objects.requireNonNull(unknownQueryField);
+    static class Result {
 
-        List<Term> extractedTerms = new ArrayList<>();
-        extractedTerms.add(new Term(unknownQueryField));
-        Fields fields = MultiFields.getFields(indexReader);
-        for (String field : fields) {
-            Terms terms = fields.terms(field);
-            if (terms == null) {
-                continue;
-            }
+        final Set<Term> terms;
+        final boolean verified;
+        final boolean noTerms;
```

Comment:
Like we discussed yesterday, we should remove the support for MatchAllDocsQuery as it complicated this code too much already. The wins are small as most percolator queries will not have this query. We should focus on the big wins or easy optimizations.